src/CowAmmModule.sol

```diff
@@ -0,0 +1,299 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.8.0 <0.9.0;
+
+import {ComposableCoW} from "lib/composable-cow/src/ComposableCoW.sol";
+import {GPv2Settlement} from "lib/composable-cow/lib/cowprotocol/src/contracts/GPv2Settlement.sol";
+import {Safe, FallbackManager, Enum} from "lib/composable-cow/lib/safe/contracts/Safe.sol";
+import {
+    ExtensibleFallbackHandler,
+    SignatureVerifierMuxer
+} from "lib/composable-cow/lib/safe/contracts/handler/ExtensibleFallbackHandler.sol";
+import {IConditionalOrder} from "lib/composable-cow/src/BaseConditionalOrder.sol";
+
+import {ConstantProduct, IPriceOracle, IERC20} from "./ConstantProduct.sol";
+
+/**
+ * @title CoW AMM Module
+ * @author CoW Protocol Developers
+ * @dev A Safe module for smoothing the experience when administering CoW AMMs.
+ */
+contract CowAmmModule {
+    // --- immutable state
+
+    /**
+     * @notice The settlement contract for CoW Protocol on this network. This is only used by the
+     * factory to determine the CowAmmModule is the correct implementation.
+     */
+    GPv2Settlement public immutable SETTLER;
+    /**
+     * @notice The EIP-712 signing domain separator for CoW Protocol on this network.
+     */
+    bytes32 public immutable COW_DOMAIN_SEPARATOR;
+    /**
+     * @notice The vault relayer that should be automatically approved for the tokens.
+     */
+    address public immutable VAULT_RELAYER;
+    /**
+     * @notice The address for the `ExtensibleFallbackHandler` on this network. This will be set as the
+     * fallback handler on the Safe when the module is attached to a Safe.
+     */
+    ExtensibleFallbackHandler public immutable EXTENSIBLE_FALLBACK_HANDLER;
+    /**
+     * @notice The address for `ComposableCoW` on this network. This will be set as the
+     * domain verifier in the `ExtensibleFallbackHandler` when the module is attached to a Safe.
+     */
+    ComposableCoW public immutable COMPOSABLE_COW;
+    /**
+     * @notice The address of the ConstantProduct programmatic order handler implementation of CoW AMMs.
+     */
+    IConditionalOrder public immutable HANDLER;
+
+    // --- mutable state
+
+    /**
+     * @notice The active CoW AMM conditional order hash (bytes32(0) if none is active).
+     */
+    mapping(Safe => bytes32) public activeOrders;
+
+    // --- constants
+
+    /**
+     * @dev The storage slot of the fallback handler on the safe
+     */
+    bytes32 internal constant _FALLBACK_HANDLER_STORAGE_SLOT =
+        0x6c9a6c4a39284e37ed1cf53d337577d14212a4870fb976a4366c693b939918d5;
```

Comment:
Is this something that is guaranteed to be static across safe's version or can we fetch it from the safe somehow on demand?

**Edit** Saw the comment below now. Let's link where this is coming from and document why this needs to be hardcoded (it's a bit of a compatibility gotcha though)